src/Abstractions/NexusMods.Abstractions.Loadouts.Synchronizers/IFileGenerator.cs

```diff
@@ -0,0 +1,24 @@
+using NexusMods.Abstractions.GameLocators;
+using NexusMods.Abstractions.Games.DTO;
+using NexusMods.Abstractions.MnemonicDB.Attributes;
+using NexusMods.Hashing.xxHash64;
+
+namespace NexusMods.Abstractions.Loadouts.Synchronizers;
+
+/// <summary>
+/// A interface for a file generator. This is used to generate the actual contents of a file during
+/// the application of a loadout.
+/// </summary>
+public interface IFileGenerator : IGuidClass
+{
+    /// <summary>
+    /// Not supported, as this is an interface
+    /// </summary>
+    /// <exception cref="NotSupportedException"></exception>
+    static UInt128 IGuidClass.Guid => throw new NotSupportedException();
+
```

Comment:
If this doesn't support `IGuidClass.Guid`, why is it implementing that interface in the first place? The interface doesn't have anything else in it.

Reply:
Maybe throw `UnreachableException` instead? Since it assumes that whatever implements `IFileGenerator` "overrides" this value, so this should never be called on `IFileGenerator`.

Reply:
Yeah it's a static member on an interface and the interface *must* implement it in that case.